lib/iris/common/metadata.py

```diff
@@ -1338,39 +1338,14 @@ def equal(self, other, lenient=None):
         return super().equal(other, lenient=lenient)
 
 
-def metadata_manager_factory(cls, **kwargs):
-    """
-    A class instance factory function responsible for manufacturing
-    metadata instances dynamically at runtime.
-
-    The factory instances returned by the factory are capable of managing
-    their metadata state, which can be proxied by the owning container.
-
-    Args:
-
-    * cls:
-        A subclass of :class:`~iris.common.metadata.BaseMetadata`, defining
-        the metadata to be managed.
-
-    Kwargs:
-
-    * kwargs:
-        Initial values for the manufactured metadata instance. Unspecified
-        fields will default to a value of 'None'.
-
-    """
-
+@lru_cache(maxsize=None)
+def _factory_cache(cls):
     def __init__(self, cls, **kwargs):
-        # Restrict to only dealing with appropriate metadata classes.
-        if not issubclass(cls, BaseMetadata):
-            emsg = "Require a subclass of {!r}, got {!r}."
-            raise TypeError(emsg.format(BaseMetadata.__name__, cls))
```

Comment:
Type checking is expensive, but thankfully we can safely remove this redundant `cls` type check at metadata manager instance creation time, as it has already been performed by the `metadata_manager_factory` function (see lines 1412-1414)

Reply:
I was considering that approach, as in, be sensible and pass in the right class otherwise all bets are off.

It is a public function but I can't ever see users calling it, and if they did, then they really need to know what their doing... so yeah, I'm happy to purge that check and the associated test. Nice one :+1: